include/sys_clock.h

```diff
@@ -28,6 +28,59 @@
 extern "C" {
 #endif
 
+/**
+ * @addtogroup clock_apis
+ * @{
+ */
+
+typedef u32_t k_ticks_t;
+
+#define K_TICKS_FOREVER ((k_ticks_t) -1)
```

Comment:
Since `k_ticks_t` is unsigned, `K_TICKS_FOREVER` is now of max integer value, instead of actual -1 (in old behavior). Would this affect the comparisons with `K_FOREVER`/`Z_FOREVER` when legacy timeout API is enabled?

Reply:
Not if everything is using the macros correctly, which is a compile error when legacy is not enabled.  The decision to make the 32 bit type unsigned and leave it that way in legacy mode was a compromise from the first version of this PR that I didn't change here.  It could be revisited I guess, but I fear this sort of thing can become a bikeshed.